kotlin/internal/jvm/compile.bzl

```diff
@@ -237,81 +314,197 @@ def kt_jvm_compile_action(ctx, rule_kind, output_jar, compile_jar):
     dirs = _compiler_directories(ctx)
     srcs = _partitioned_srcs(ctx.files.srcs)
     friend = _compiler_friends(ctx, friends = getattr(ctx.attr, "friends", []))
-    compile_deps = _compiler_deps(toolchains, friend, deps = ctx.attr.deps + ctx.attr.plugins)
+    compile_deps = _compiler_deps(toolchains, friend, deps = ctx.attr.deps)
     annotation_processors = _plugin_mappers.targets_to_annotation_processors(ctx.attr.plugins + ctx.attr.deps)
     transitive_runtime_jars = _plugin_mappers.targets_to_transitive_runtime_jars(ctx.attr.plugins + ctx.attr.deps)
     plugins = ctx.attr.plugins
 
-    if toolchains.kt.experimental_use_abi_jars:
-        kt_compile_jar = ctx.actions.declare_file(ctx.label.name + "-kt.abi.jar")
+    output_jars = []
+    if toolchains.kt.experimental_use_java_builder:
```

Comment:
Pull this into a separate function? It's long enough before forking the new way and the old way.

Reply:
Done.